File: OC-BlockDemo/OC-BlockDemo/ViewController.m
//
//  ViewController.m
//  OC-BlockDemo
//
//  Created by 邱学伟 on 16/10/20.
//  Copyright © 2016年 邱学伟. All rights reserved.
//

#import "ViewController.h"
#import "HttpTool.h"
@interface ViewController ()
@property (nonatomic, strong) HttpTool *tool;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.tool = [[HttpTool alloc] init];
    
}

-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    [self.tool requestDate:^{
        
        NSLog(@"返回服务器数据更新UI___:%@",[NSThread currentThread]);
        
        self.view.backgroundColor = [UIColor redColor];
    }];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SwiftDemo/SwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SwiftDemo
//
//  Created by 邱学伟 on 16/10/20.
//  Copyright © 2016年 邱学伟. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        //类的函数重载
//        let person = Person()
//        let sumDoule : Double = person.sum(num1: 1.9, num2: 2.2)
//        print(sumDoule)
//        print(person.sum(num1: 1, num2: 2, num3: 3))
        
        //swift 中的block
        let httpTool : HttpToolSwift = HttpToolSwift()
        httpTool.requestData { 
            //返回服务器获取的数据,更新UI
            print("返回服务器获取的数据,更新UI...当前线程:\(Thread.current)")
            self.view.backgroundColor = UIColor.red
        }
        
    }

}

